server/app/repository/ApplicationRepository.java

```diff
@@ -118,6 +127,15 @@ private Application submitApplicationInternal(
             || app.getLifecycleStage().equals(LifecycleStage.OBSOLETE)) {
           continue;
         }
+        LOGGER.error(
+            "Multiple applications found at submit time for applicant {} to program {} {}:"
+                + " application {}",
+            applicant.id,
+            program.id,
+            program.getProgramDefinition().adminName(),
+            app.id);
```

Comment:
(NB) I'd also consider making this error a warning since it doesn't seem to create a hard stop for the applicant.

Reply:
done